client/src/main/java/io/pravega/client/netty/impl/ConnectionFactoryImpl.java

```diff
@@ -57,14 +58,14 @@
     private final AtomicBoolean closed = new AtomicBoolean(false);
     private final ScheduledExecutorService executor = ExecutorServiceHelpers.newScheduledThreadPool(POOL_SIZE,
                                                                                                     "clientInternal");
+    private final String certFile;
 
     /**
      * Actual implementation of ConnectionFactory interface.
-     *
-     * @param ssl Whether connection should use SSL or not.
      */
-    public ConnectionFactoryImpl(boolean ssl) {
-        this.ssl = ssl;
+    public ConnectionFactoryImpl() {
+        this.ssl = Boolean.parseBoolean(System.getProperty("io.pravega.auth.enabled"));
```

Comment:
Earlier I made a comment about how if auth is enabled, it makes sense to assume this is over ssl. (else this means sending credentials over in plain text which doesn't sound good).

But...  I'm now wondering if having the flexibility to keep this two toggles would be better.   Consider the case where one will deploy the controllers behind a load balancer and terminating SSL there.   It might be nice, in some cases, to have auth enabled on the controller and segment stores, but not SSL in this case.

It would also help implementors/deployers tackle authentication and certificate management separately.  I'm currently experimenting with this with DCOS/Mesos frameworks and while having to deal with segment stores getting their own certs and managing a cluster CA is something that I will have to do no matter what, it would be nice like I said to be able to tackle these two problems separately.

What do you think?